controller/service.go

```diff
@@ -123,6 +136,25 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.clearServiceState(key, svc)
 			lbIPs = []net.IP{}
 		}
+		if len(desiredLbIPs) > 0 {
+			desiredPool, err = c.ips.GetPoolOwner(desiredLbIPs)
+			if err != nil {
+				level.Error(l).Log("event", "loadbalancerIP", "error", err, "msg", "invalid ip pool for requested loadbalancer IPs")
+				c.client.Errorf(svc, "LoadBalancerFailed", "invalid ip pool for requested loadbalancer IPs: %s", err)
+				c.clearServiceState(key, svc)
+				return true
+			}
+		}
+
+		// Check if desired ip pool is compatible for assigning ip for the service.
+		// If it is not, exit immediately with an error.
+		if desiredPool != nil {
+			if desiredPool.DisablePoolSelection || !isPoolCompatibleWithService(desiredPool, svc) {
```

Comment:
just a thought: with this logic, there's no need for the flag right?
Because what is going to happen is, a service is able to pin to a pool only if the pool selects it.

Reply:
yes, as discussed offline, removed flag now from ip pool crd and doing only ip pool compatibility check for a given service during ip assignment.